### PR TITLE
feat: what is PoP, what it can be

### PR DESCRIPTION
**Pull request by peppelinux**
this PR closes https://github.com/peppelinux/draft-demarco-oauth-status-attestations/issues/11

### CODE REVIEW

File: draft-demarco-oauth-status-attestations.md [hunk 1]
```diff
@@ -215,6 +215,16 @@ format may be used, as long as all attributes and requirements
...
+
+While this specification does not prescribe any additional methods for the PoP of the credential, it aims to offer guidance for concrete implementations utilizing common PoP mechanisms. This includes, but is not limited to:
+
+1. Having the digital representation of the credential (the bytes).
+2. Controlling a private key that corresponds to a public key associated with the credential, often indicated within the credential's cnf (confirmation) claim or through a similar mechanism.
+
+The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the credential can execute actions exclusively reserved for the legitimate owner. This dual-layered approach not only reinforces the security and integrity of the status attestation process but also affirms that the entity requesting a status attestation is the authentic holder of the credential.
```
Comment (fmarino-ipzs) [inline draft-demarco-oauth-status-attestations.md:227]:
```suggestion
The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the credential can execute actions exclusively reserved for the legitimate owner. This dual-layered approach not only reinforces the security and integrity of the status attestation process but also affirms that the entity requesting a status attestation is the same entity to which the credential was issued. 
```

**Review by fmarino-ipzs — APPROVED** [review]
LGTM.
Btw, we are assuming here that the credential for which a Status Attestation is requested has a cryptographic holder-binding. Is this a right assumption? Should we generalize to cases where we have other holder binding mechanisms or even none?

File: draft-demarco-oauth-status-attestations.md (hunk 1 above)
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:227]:
```suggestion
The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the credential can execute actions exclusively reserved for the legitimate owner. The dual-layered approach of requiring both possession of the credential and control over the corresponding private key indeed reinforces the security and integrity of the status attestation process. It also ensures that the entity requesting a status attestation is indeed the same entity to which the credential was originally issued, affirming the authenticity and rightful ownership of the credential.
```

**Comment by peppelinux** [comment]
@fmarino-ipzs This pull request addresses a limitation of the current specification. It currently only works with digital credentials that contain holder key binding.

File: draft-demarco-oauth-status-attestations.md [hunk 2]
```diff
@@ -215,6 +215,16 @@ format may be used, as long as all attributes and requirements
...
+
+While this specification does not prescribe any additional methods for the PoP of the credential, it aims to offer guidance for concrete implementations utilizing common PoP mechanisms. This includes, but is not limited to:
+
+1. Having the digital representation of the credential (the bytes).
+2. Controlling a private key that corresponds to a public key associated with the credential, often indicated within the credential's cnf (confirmation) claim or through a similar mechanism.
+
+The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the credential can execute actions exclusively reserved for the legitimate owner. The dual-layered approach of requiring both possession of the credential and control over the corresponding private key indeed reinforces the security and integrity of the status attestation process. It also ensures that the entity requesting a status attestation is indeed the same entity to which the credential was originally issued, affirming the authenticity and rightful ownership of the credential.
```
Comment (OR13) [inline draft-demarco-oauth-status-attestations.md:227]:
```suggestion
The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the credential can execute actions exclusively reserved for the legitimate subject. The dual-layered approach of requiring both possession of the credential and control over the corresponding private key indeed reinforces the security and integrity of the status attestation process. It also ensures that the entity requesting a status attestation is indeed the same entity to which the credential was originally issued, affirming the authenticity and rightful possession of the credential.
```

**Review by OR13 — CHANGES_REQUESTED** [review]
minor word changes, ideally we don't need to distinguish "owner" and "subject", and "ownership" and "possession".

File: draft-demarco-oauth-status-attestations.md [hunk 3]
```diff
@@ -215,6 +215,16 @@ format may be used, as long as all attributes and requirements
 defined in this specification are satisfied, even using equivalent names
 or values.
 
+# Proof of Possession of a Credential
+
+The concept of Proof of Possession (PoP) of a Credential within the framework of the Status Attestation specification encompasses a broader perspective than merely possessing the digital bytes of the credential. It involves demonstrating rightful control or ownership over the credential, which can manifest in various forms depending on the technology employed and the nature of the digital credential itself. For instance, a credential could be presented visually (de-visu) with a personal portrait serving as a binding element.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:220]:
```suggestion
The concept of Proof of Possession (PoP) of a Credential within the framework of the Status Attestation specification encompasses a broader perspective than merely possessing the digital bytes of the Credential. It involves demonstrating rightful control or ownership over the Credential, which can manifest in various forms depending on the technology employed and the nature of the digital Credential itself. For instance, a Credential could be presented visually (de-visu) with a personal portrait serving as a binding element.
```

File: draft-demarco-oauth-status-attestations.md [hunk 4]
```diff
@@ -215,6 +215,16 @@ format may be used, as long as all attributes and requirements
...
+
+While this specification does not prescribe any additional methods for the PoP of the credential, it aims to offer guidance for concrete implementations utilizing common PoP mechanisms. This includes, but is not limited to:
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:222]:
```suggestion
While this specification does not prescribe any additional methods for the proof of possession of the Credential, it aims to offer guidance for concrete implementations utilizing common proof of possession mechanisms. This includes, but is not limited to:
```

File: draft-demarco-oauth-status-attestations.md [hunk 5]
```diff
@@ -215,6 +215,16 @@ format may be used, as long as all attributes and requirements
...
+
+1. Having the digital representation of the credential (the bytes).
+2. Controlling a private key that corresponds to a public key associated with the credential, often indicated within the credential's cnf (confirmation) claim or through a similar mechanism.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:225]:
```suggestion
2. Controlling a private key that corresponds to a public key associated with the Credential, often indicated within the Credential's cnf (confirmation) claim or through a similar mechanism.
```

File: draft-demarco-oauth-status-attestations.md [hunk 6]
```diff
@@ -215,6 +215,16 @@ format may be used, as long as all attributes and requirements
...
+
+The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the credential can execute actions exclusively reserved for the legitimate subject. The dual-layered approach of requiring both possession of the credential and control over the corresponding private key indeed reinforces the security and integrity of the status attestation process. It also ensures that the entity requesting a status attestation is indeed the same entity to which the credential was originally issued, affirming the authenticity and rightful possession of the credential.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-attestations.md:227]:
```suggestion
The essence of requiring control over the private key and its demonstration through a cryptographic operation (e.g., signing a challenge or a token) is to ensure that the entity in possession of the Credential can execute actions exclusively reserved for the legitimate subject. The dual-layered approach of requiring both possession of the Credential and control over the corresponding private key indeed reinforces the security and integrity of the status attestation process. It also ensures that the entity requesting a Status Attestation is indeed the same entity to which the Credential was originally issued, affirming the authenticity and rightful possession of the Credential.
```

**Review by OR13 — APPROVED** [review]

**Pull request merged**